src/azure-cli/azure/cli/command_modules/storage/oauth_token_util.py

```diff
@@ -29,13 +32,15 @@ def timer_callback(self):
             resource="https://storage.azure.com", subscription=self.cli_ctx.data['subscription_id'])[0][2]
         try:
             self.token_credential.token = token['accessToken']
+            expire = token['expiresOn']
             seconds_left = (datetime.strptime(token['expiresOn'], "%Y-%m-%d %H:%M:%S.%f") - datetime.now()).seconds
         except KeyError:  # needed to deal with differing unserialized MSI token payload
             self.token_credential.token = token['access_token']
+            expire = datetime.datetime.fromtimestamp(int(token['expires_on']))
             seconds_left = (datetime.fromtimestamp(int(token['expires_on'])) - datetime.now()).seconds
+
         if seconds_left < 180:
-            # acquired token expires in less than 3 mins
-            raise Exception("Acquired a token expiring in less than 3 minutes")
+            logger.warning("Acquired token will expire on %s.", expire)
```

Comment:
May I ask why does `seconds_left` compare with 180s, but the error is `will expire on %s.", expire`? Why not `seconds_left < expire`

Reply:
expire is actual time and seconds left is time range. The compare is used to make sure we there is enough buffer time for the acquired token to be used in the request.